### ISSUE TITLE
Trying to use in OSX

### ISSUE THREAD

**Issue by takaxp** [Ambrevar/emacs-gif-screencast#4]
Hi.

I'm trying to use this pretty nice elisp to get an animation for demonstrating actions in Emacs. But the capture command `scrot` is not familiar to me since I'm OSX user.

For OSX user, `/usr/sbin/screencapture` is the default and useful command to take screenshots by CUI. So I made some changes on the original code like [this](https://github.com/takaxp/emacs-gif-screencast/commit/64a5d3dcf4115aee2fed5b8bf5877b4348732240#diff-825716ca58bd0ebef62b7ed711446577). Unfortunately, `screencapture` will not crop the region for an Emacs frame in the captured screen, then every captured image should be cropped just before generating GIF file. PPM format is required for capturing images since PNG does not work for the approach of cropping images by `mogfiry` command when generating GIF from PNG files.

With the following setting, the modified code works for me.

```
(when (require 'gif-screencast nil t)
  (setq gif-screencast-want-optimized nil)
  (setq gif-screencast-program "/usr/sbin/screencapture")
  (setq gif-screencast-capture-format "ppm")
  (setq gif-screencast-args '("-x")) ;; shut-up shutter sounds
  (setq gif-screencast-autoremove-screenshots t) ;; I'm not sure autoremove is actually available in the original code.
  (defun advice:gif-screencast-stop ()
    "Open the output directory when screencast is finished."
    (shell-command-to-string
     (concat "open " gif-screencast-output-directory)))
  (advice-add 'gif-screencast-stop :after #'advice:gif-screencast-stop))
```

I'm happy to create a PR, but I believe you are also trying to adopt `async` library. So I hold it by finishing your work.

Any comments or suggestions on this?

Best,
Takaaki

**Comment by Ambrevar** [comment]
Excellent!  Thank you for you hard work, this is a fantastic contribution.

I don't think there is any need for `async`, but PR #2 will fix a race condition which significantly refactors the code, so you might want to wait until that's merged (very soon).

What do you mean with
```
  (setq gif-screencast-autoremove-screenshots t) ;; I'm not sure autoremove is actually available in the original code.
```
?  Not working?

**Comment by takaxp** [comment]
OK, I'll wait for the completion of PR #2.

For me, `gif-screencast-autoremove-screenshots` did not work when `gif-screencast-want-optimized` was `nil`.
We can see the flag in the last part of your code.  I believe `(process-exit-status p)` will not reflect the correct status if `gif-screencast-want-optimized` is `nil` since the optimization process will be skipped.

So I wrapped `delete-file` commands by `set-process-sentinel` again. This will ensure the file deletion after the completion of GIF converting process. At least, I confirmed this in OSX.

```
      (if gif-screencast-autoremove-screenshots
          (set-process-sentinel p
                                (lambda (process event)
                                  (if (and (eq (process-status process) 'exit)
                                           (= (process-exit-status process) 0))
                                      (dolist (f gif-screencast--frames)
                                        (delete-file (cdr f))))
                                  (gif-screencast-print-status process event)))
        (set-process-sentinel p 'gif-screencast-print-status))
```

I'll check this issue again and create a separate PR from the OSX support if needed.

Best,
Takaaki

**Comment by Ambrevar** [comment]
Ouch, that was a big oversight on my end! :/
I've just merged #2, so feel free to send a PR will all you changes.

**Comment by takaxp** [comment]
Hi!

I just created a PR for OSX support.
Please check the proposed changes described in #5.

Best,
Takaaki
